app/models/project_services/composer_service.rb

```diff
@@ -0,0 +1,516 @@
+              ['WordPress Plugin', 'wordpress-plugin'],
+              ['WordPress Theme', 'wordpress-theme'],
+              ['Zend Extra', 'zend-extra'],
+              ['Zend Library', 'zend-library'],
+              ['Zend Module', 'zend-module'],
+              ['Zikula Module', 'zikula-module'],
+              ['Zikula Theme', 'zikula-theme']
+            ],
+            default_choice: 'library'
+          },
+          { type: 'text',
+            name: 'custom_package_type',
+            title: 'Custom Package Type',
+            help: 'A customized package type to use.'
+          }
+        ]
+      },
+      { type: 'fieldset',
+        legend: 'Advanced Packaging Mode Settings',
+        fields:
+        [
+          { type: 'textarea',
+            name: 'custom_json',
+            title: 'Custom JSON',
+            help: 'The custom json to used to export this project as a composer package.'
+          }
+        ]
+      },
+      { type: 'fieldset',
+        legend: 'Branch Selection',
+        fields:
+        [
+          { type: 'checkbox',
+            name: 'export_branches',
+            title: 'Export',
+          },
+          { type: 'text',
+            name: 'branch_filters',
+            title: 'Filters',
+            help: 'Separate branches with commas. '\
+                  'Leave blank to export all branches.'
+          }
+        ]
+      },
+      { type: 'fieldset',
+        legend: 'Tag Selection',
+        fields:
+        [
+          { type: 'checkbox',
+            name: 'export_tags',
+            title: 'Export'
+          },
+          { type: 'text',
+            name: 'tag_filters',
+            title: 'Filters',
+            help: 'Separate tags with commas. '\
+                  'Leave blank to export all tags.'
+          }
+        ]
+      }
+    ]
+  end
+
+  # disable test button
+  def can_test?
+    false
+  end
+
+  def supported_events
+    %w(push tag_push)
+  end
+
+  def execute(data)
+    return unless supported_events.include?(data[:object_kind])
+    process_project
+  end
+
+  def process_project
+
+    # do not process when service template
+    return if template?
+
+    # delete the project repository file since it will be regenerated.
+    File.delete(repo_path) if File.exists?(repo_path)
+
+    if activated?
+      # process packages for all tags
+      project.repository.tags.each do |tag|
+        process_commit(tag)
+      end
+
+      # process packages for all branches
+      project.repository.branches.each do |branch|
+        process_commit(branch)
+      end
+
+      # write the repository json file.
+      repository.write
+    end
+
+    # update the root json file to include/exlude this project.
+    update_root_file
+
+  end
+
+  def process_commit(ref)
+    if activated? && ref_exported?(ref)
+      package = ref_package(ref)
+      repository.add_package(package)
+    end
+  rescue Exception => e
```

Comment:
Please don't rescue from `Exception`: http://stackoverflow.com/questions/10048173/why-is-it-bad-style-to-rescue-exception-e-in-ruby